script/github-actions/get-changed-apps.js

```diff
@@ -4,84 +4,91 @@ const find = require('find');
+ * @returns {Object[]|null} Sliced manifests of allowed apps. Otherwise null.
  */
-const getAllowedApp = (file, allowList, outputType = 'entry') => {
-  if (!file.startsWith('src/applications')) return null;
-
-  const manifest = getManifest(file);
-  const entryName = manifest?.entryName;
-
-  if (allowList.includes(entryName)) {
-    // Return the entry name, folder path, or root URL depending on the output type
-    if (outputType === 'entry') {
-      return entryName;
-    } else if (outputType === 'folder') {
-      const appFolderName = file.split('/')[2];
-      return `src/applications/${appFolderName}`;
-    } else if (outputType === 'url') {
-      return manifest?.rootUrl || '';
-    } else throw new Error('Invalid output type specified.');
+const getAllowedApps = (filePath, allow) => {
+  const appsDirectory = 'src/applications';
+
+  if (!filePath.startsWith(appsDirectory)) return null;
+
+  const rootAppFolder = filePath.split('/')[2];
+  const rootAppPath = path.join(appsDirectory, rootAppFolder);
+  const manifests = getManifests(filePath);
+
+  const isAllowed =
+    allow.rootAppFolders.includes(rootAppFolder) ||
+    (manifests.length === 1 &&
+      allow.entryNames.includes(manifests[0].entryName));
+
+  if (isAllowed) {
+    return manifests.map(({ entryName, rootUrl }) => ({
+      entryName,
+      rootUrl,
+      rootPath: rootAppPath,
+    }));
   }
 
   return null;
 };
 
 /**
  * Checks if a changed apps build is possible by confirming that all
- * files are from apps on an allow list. If so, returns a comma-delimited list
- * of app entry names, relative paths, or URLs. If not, returns an empty string.
+ * files are from apps on an allow list. If so, returns a comma-delimited string
+ * of app entry names, relative paths, or URLs; otherwise returns an empty string.
  *
- * @param {string[]} files - An array of relative file paths.
+ * @param {string[]} filePaths - An array of relative file paths.
  * @param {Object} config - The changed apps build config.
  * @param {string} outputType - Determines what app information should be returned.
  * @returns {string} A comma-delimited string of app entry names, relative paths or URLs.
  */
-const getChangedAppsString = (files, config, outputType = 'entry') => {
-  const allowedApps = [];
-
-  for (const file of files) {
-    const allowedApp = getAllowedApp(file, config.allow, outputType);
-
-    if (allowedApp !== null) {
-      // Some allowed apps don't have URLs or other attributes, so only add to
-      // the list when the string of the allowed app's output type isn't empty
-      if (allowedApp.length) allowedApps.push(allowedApp);
-    } else {
-      return '';
-    }
+const getChangedAppsString = (filePaths, config, outputType = 'entry') => {
+  const appStrings = [];
+
+  for (const filePath of filePaths) {
+    const allowedApps = getAllowedApps(filePath, config.allow);
+
+    if (allowedApps) {
+      allowedApps.forEach(app => {
```

Comment:
Cool, it makes sense to do the outputType mapping here anyway since this function is building the string.

Reply:
Right, this way we also don't need to pass the output type into two different functions